src/hb-priority-queue.hh

```diff
@@ -42,10 +42,11 @@
  * priority of its children. The heap is stored in an array, with the
  * children of node i stored at indices 2i + 1 and 2i + 2.
  */
+template <typename Type>
```

Comment:
Can you call `Type` just `K` or `Key`? We might want to templatize the value one day as well.